src/libdrakvuf/drakvuf.c

```diff
@@ -585,7 +585,11 @@ int drakvuf_get_address_width(drakvuf_t drakvuf)
 bool drakvuf_get_current_process_data(drakvuf_t drakvuf, drakvuf_trap_info_t* info, proc_data_priv_t* proc_data)
 {
     addr_t process_base = drakvuf_get_current_process(drakvuf, info);
-    return drakvuf_get_process_data_priv(drakvuf, process_base, proc_data);
+    if ( drakvuf_get_os_type(drakvuf) == VMI_OS_LINUX)
```

Comment:
Instead of this if-else you should just have a call to both `drakvuf_get_process_data_priv` and `drakvuf_get_current_thread_id`regardless of the OS.